flow/aggregates.py

```diff
@@ -0,0 +1,463 @@
+        :type reverse_order:
+            bool
+        :param select:
+            Condition for filtering individual jobs. This is passed as the
+            callable argument to `filter`.
+            The default behavior is no filtering.
+        :type select:
+            callable or NoneType
+        """
+        if isinstance(key, str):
+            if default is None:
+                def keyfunction(job):
+                    return job.sp[key]
+            else:
+                def keyfunction(job):
+                    return job.sp.get(key, default)
+        elif isinstance(key, Iterable):
+            keys = list(key)
+            if default is None:
+                def keyfunction(job):
+                    return [job.sp[key] for key in keys]
+            else:
+                if isinstance(default, Iterable):
+                    if len(default) != len(keys):
+                        raise ValueError("Expected length of default argument is "
+                                         f"{len(keys)}, got {len(default)}.")
+                else:
+                    raise TypeError("Invalid default argument. Expected Iterable, "
+                                    f"got {type(default)}")
+
+                def keyfunction(job):
+                    return [job.sp.get(key, default_value)
+                            for key, default_value in zip(keys, default)]
+        elif callable(key):
+            keyfunction = key
+        else:
+            raise TypeError("Invalid key argument. Expected either str, Iterable "
+                            f"or a callable, got {type(key)}")
+
+        def aggregator(jobs):
+            for key, group in groupby(sorted(jobs, key=keyfunction), key=keyfunction):
+                yield group
+
+        return cls(aggregator, sort_by, reverse_order, select, key=key, default=default)
+
+    def __eq__(self, other):
+        if type(self) != type(other):
+            return False
+        elif(
+            self._sort_by != other._sort_by or
+            self._reverse_order != other._reverse_order or
+            self._kwargs != other._kwargs
+        ):
+            return False
+        elif(
+            self._select == other._select and
+            self._aggregator == other._aggregator
+        ):
+            return True
+
+        # Get unique id for _select attribute
+        self_select = self._get_unique_function_id(self._select)
+        other_select = self._get_unique_function_id(other._select)
+        # Get unique id for _aggregator attribute
+        self_aggregator = self._get_unique_function_id(self._aggregator)
+        other_aggregator = self._get_unique_function_id(other._aggregator)
+
+        if(
+            self_select == other_select and
+            self_aggregator == other_aggregator
+        ):
+            return True
+
+        return False
+
+    def __hash__(self):
+        blob_l = list(map(hash, [self._sort_by, self._reverse_order]))
```

Comment:
Using map for these two attributes seems a bit much. Just writing it explicitly would be fine.